sparse/__init__.py

```diff
@@ -7,3 +7,9 @@
 from ._version import get_versions
 __version__ = get_versions()['version']
 del get_versions
+
+import os
+
+_AUTO_DENSIFICATION_ENABLED = bool(int(os.environ.get('SPARSE_AUTO_DENSIFY', '1')))
+
```

Comment:
I would also add

```py
def set_auto_densify(flag):
    global _AUTO_DENSIFICATION_ENABLED
    _AUTO_DENSIFICATION_ENABLED = flag
```

Reply:
The problem is that that isn't thread safe. For example, if you do it with Dask or whatever else, it can blow stuff up.

An environment variable loaded at import time is guaranteed to be thread safe.

Reply:
Can we add a warning to the function that it isn't thread safe. I think if it is only called right after import it should be fine too.

Reply:
> unfortunately there's no clear way to make it user facing but not library facing.

That's in the same category as `numpy.set_printoptions`, `np.seterr`, private class attributes, etc. - just don't mess with it as a library. It doesn't happen a lot, and if it does the first person who notices files a bug against that library.